src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -1292,7 +1292,11 @@ std::future<data::ResultCode::Numeric> SotaUptaneClient::sendFirmwareAsync(Uptan
     sendEvent<event::InstallStarted>(secondary.getSerial());
     report_queue->enqueue(std_::make_unique<EcuInstallationStartedReport>(secondary.getSerial(), correlation_id));
 
-    data::ResultCode::Numeric result = secondary.install(target);
+    data::ResultCode::Numeric result = secondary.sendFirmware(target);
```

Comment:
Not sure if I see any benefits in having two calls (sendFirmware and install) instead of just one `install` call in this context, although it doesn't cause any issue.

Reply:
After some discussion, we decided it was best to split it, because we ultimately want to move the `sendFirmware` part to the download phase on the Primary. This is just the first step to enable that.

Reply:
>  we ultimately want to move the sendFirmware part to the download phase on the Primary.

That's exactly what I'd considered too and that's why I suggested adding download() method to the secondary interface in that confluence document. In this case, a secondary implementer can implement streaming to Secondary ECU, provided that SotaUptaneClient calls ecu/secondary->download() at a download phase so the implementer just calls corresponding send data RPC call in its implementation, and sendFirmware is not required in the interface it is just needed at the RPC API level. But, it's just my IMHO :), and I don't see any issue with having sendFirmware in the interface neither.

Reply:
Thanks for your input! Sounds like your download concept is pretty similar to my intention for the sendFirmware once we move it. It's still up to the implementer to decide what to do in the download/sendFirmware call and what to do in the install call, and I imagine some will skip the sendFirmware part, especially if the Secondary doesn't have space to keep the update on disk. But I want to have the option for it if someone wants or needs it.

Reply:
:+1: